geotrek/sensitivity/models.py

```diff
@@ -296,6 +300,41 @@ def pretty_practices(self):
                                lambda self: intersecting(tourism_models.TouristicEvent, self, 0).filter(published=True),
                                _("Published touristic events"))
 
+
+if 'geotrek.outdoor' in settings.INSTALLED_APPS:
+    from geotrek.outdoor import models as outdoor_models
+
+    outdoor_models.Site.add_property('sensitive_areas',
+                                     lambda self: intersecting(SensitiveArea, self, distance=0,
+                                                               ordering=False, field='geom_buffered'),
+                                     _("Sensitive areas"))
+    outdoor_models.Site.add_property('published_sensitive_areas',
+                                     lambda self: intersecting(SensitiveArea, self, distance=0, ordering=False,
+                                                               field='geom_buffered').filter(published=True),
+                                     _("Published sensitive areas"))
+    outdoor_models.Course.add_property('sensitive_areas',
+                                       lambda self: intersecting(SensitiveArea, self, distance=0,
+                                                                 ordering=False, field='geom_buffered'),
+                                       _("Sensitive areas"))
+    outdoor_models.Course.add_property('published_sensitive_areas',
+                                       lambda self: intersecting(SensitiveArea, self, distance=0, ordering=False,
+                                                                 field='geom_buffered').filter(published=True),
+                                       _("Published sensitive areas"))
+
+    SensitiveArea.add_property('sites',
+                               lambda self: intersecting(outdoor_models.Site, self),
+                               _("Touristic contents"))
+    SensitiveArea.add_property('published_sites',
+                               lambda self: intersecting(outdoor_models.Site, self).filter(published=True),
+                               _("Published touristic contents"))
+    SensitiveArea.add_property('courses',
+                               lambda self: intersecting(outdoor_models.Course, self),
+                               _("Touristic events"))
+    SensitiveArea.add_property('published_courses',
+                               lambda self: intersecting(outdoor_models.Course, self).filter(published=True),
+                               _("Published touristic events"))
```

Comment:
you should add outdoor related properties in outdoor models.py

Reply:
Do you mean to replace lambdas with actual methods on outdoor models?

Reply:
We have discussed the matter. The `sensitive_areas` properties cannot be added to Site and Course in the `geotrek.outdoor.models` module because of a circular import (`geotrek.sensitivity.models` already imports the outdoor module so the outdoor module cannot import sensitivity to get the `SensitiveArea` class).